### PR TITLE
fix(provider/cf): fix route definition for CF manifest deployment

### PR DESCRIPTION
**Pull request by jmelchio**
Co-authored-by: Jon Schneider <jschneider@pivotal.io>

### CODE REVIEW

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/model/v3/CreateApplication.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import lombok.Getter;
 
 import javax.annotation.Nullable;
+import javax.validation.constraints.Null;
```
Comment (jkschneider) [inline clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/model/v3/CreateApplication.java]: Extraneous import that can be deleted.

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/converters/DeployCloudFoundryServerGroupAtomicOperationConverter.java [hunk 2]
```diff
@@ -111,21 +119,72 @@ public DeployCloudFoundryServerGroupDescription convertDescription(Map input) {
...
-            });
-
-        apps.stream().findFirst().ifPresent(converted::setApplicationAttributes);
+          .convertValue(manifestMap.get("applications"), new TypeReference<List<CloudFoundryManifest>>() {
+          });
+
+        manifestApps.stream().findFirst().ifPresent(app -> {
+          DeployCloudFoundryServerGroupDescription.ApplicationAttributes attrs = convertBaseAttributes(app);
+          attrs.setRoutes(app.getRoutes() == null ? null : app.getRoutes().stream().flatMap(route -> route.values().stream()).collect(toList()));
+          attrs.setEnv(app.getEnv() == null ? null : app.getEnv().stream().flatMap(env -> env.entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
```
Comment (jkschneider) [inline clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/converters/DeployCloudFoundryServerGroupAtomicOperationConverter.java]: static import `Collectors.toMap`

**Comment by jmelchio** [comment]
@jkschneider the ToOneRelationship is there for mapping purposes. Shouldn't we leave that in place for serialization and de-serialization?

**Comment by jkschneider** [comment]
> the ToOneRelationship is there for mapping purposes. Shouldn't we leave that in place for serialization and de-serialization?

I was referring to `ApplicationAttributes#env`, not `CreateApplication#relationships`.

**Pull request merged**